posthog/hogql_queries/query_runner.py

```diff
@@ -313,8 +335,12 @@ def run(self, refresh_requested: Optional[bool] = None) -> CachedQueryResponse:
                     return cached_response
                 else:
                     QUERY_CACHE_HIT_COUNTER.labels(team_id=self.team.pk, cache_hit="stale").inc()
+                    if execution_mode == ExecutionMode.CACHE_ONLY:
+                        return cached_response
```

Comment:
Do we need to set `is_cached = True` here as well? 🤔

Reply:
Hmm, yes, reworked and hopefully clarified this code overall